### ISSUE TITLE
Keep champions between generations

### ISSUE THREAD

**Issue by qwertyuu** [yaricom/goNEAT#47]
Suggestion in the title. Add a configuration to keep champions (N best of generation, or N best of each species) between generations, so we can keep the best organism and not lose their "winner" genome in breeding.

**Comment by yaricom** [comment]
Yes, this is interesting idea, which requires further investigation.
However, it has some drawbacks:
- At each epoch of evolution it is hard to estimate which organism will lead to the winner. Even if organism not the best right now, it can have some beneficial mutation which can produce winner later. On the other hand, current champion can be stale without further potential.
- We have limited size of population. If we are going to keep N champions at each epoch this can broke the evolutionary process by having in population only stale (senile) organisms at the later stages of evolution.

But, anyway, this is good idea and it can be studied further. 

Thank you for suggestion.

**Comment by qwertyuu** [comment]
Yes you are right, keeping the current champion biases the search. I saw it first in BoxCar2D which is in flash, but was re-used in a HTML5-canvas copy here: https://rednuht.org/genetic_cars_2/ 

It is called "Elite clones".

Thanks for taking time to think of this with me! I might try my hand at coding some of my issues if I think I understand your engine good enough
